packages/loader/container-loader/src/container.ts

```diff
@@ -1438,7 +1438,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             this.logger.sendTelemetryEvent({
                 eventName:"NoCodeProposal",
             });
-            await new Promise<void>((resolve)=>{
+            // eslint-disable-next-line @typescript-eslint/no-floating-promises
+            new Promise<void>((resolve)=>{
```

Comment:
this is not safe to do here

Reply:
Yes, wanted to know why? So we should not return container before code proposal is what we are saying here?

Reply:
that and i don't think we can continue here

Reply:
the code is pointless if we don't wait for it.

Reply:
It creates a deadlock because we are stopping protocol handler to be initialized and then when the code proposal message will be received, then protocol handler needs to process that which is not yet initialized.

Reply:
where does it block?

Reply:
ok. this is what i think we should do. take the next part and put it into a function:
```
const addProposalHandlers = ()=>{


<!--StartFragment-->        protocol.quorum.on("addProposal", (proposal: IPendingProposal) =&gt; {            if (proposal.key === "code" || proposal.key === "code2") {                this.emit("codeDetailsProposed", proposal.value, proposal);            }        });        protocol.quorum.on(            "approveProposal",            (sequenceNumber, key, value) =&gt; {                debug(`approved ${key}`);                if (key === "code" || key === "code2") {                    debug(`codeProposal ${JSON.stringify(value)}`);                    if (!isFluidCodeDetails(value)) {                        this.logger.sendErrorEvent({                                eventName: "CodeProposalNotIFluidCodeDetails",                        });                    }                    this.processCodeProposal().catch((error) =&gt; {                        this.close(CreateContainerError(error));                        throw error;                    });                }            });<!--EndFragment-->


}
```
then the call that add the end of the promise and then we can fire and forget. then add an else case where we just call it (same as now). i think that will work.